web/src/containers/ApdKeyPersonnel.js

```diff
@@ -0,0 +1,165 @@
+              </tr>
+            </thead>
+            <tbody>
+              {keyPersonnel.map((d, i) => (
+                <tr key={d.id}>
+                  <td className="mono">{i + 1}.</td>
+                  <td>
+                    <Input
+                      name={`state-person-${d.id}-title`}
+                      label={t('apd.keyPersonnel.labels.title')}
+                      hideLabel
+                      value={d.title}
+                      onChange={this.handleChange(i, 'title')}
+                    />
+                  </td>
+                  <td>
+                    <Textarea
+                      name={`state-person-${d.id}-desc`}
+                      label={t('apd.keyPersonnel.labels.desc')}
+                      hideLabel
+                      rows="3"
+                      value={d.desc}
+                      onChange={this.handleChange(i, 'desc')}
+                    />
+                  </td>
+                  {years.map(year => (
+                    <Fragment key={year}>
+                      <td>
+                        <Input
+                          name={`state-person-${d.id}-${year}-amt`}
+                          label={t('apd.keyPersonnel.labels.costAmt')}
+                          hideLabel
+                          value={d.years[year].amt}
+                          onChange={this.handleChange(i, 'amt', year)}
+                        />
+                      </td>
+                      <td>
+                        <Input
+                          name={`state-person-${d.id}-${year}-amt`}
+                          label={t('apd.keyPersonnel.labels.costPerc')}
+                          hideLabel
+                          value={d.years[year].perc}
+                          onChange={this.handleChange(i, 'perc', year)}
+                        />
+                      </td>
+                    </Fragment>
+                  ))}
+                  <td className="center">
+                    <button
+                      type="button"
+                      className="btn btn-outline border-silver px1 py-tiny mt-tiny"
+                      title={t('apd.keyPersonnel.removeLabel')}
+                      onClick={() => removePerson(d.id)}
+                    >
+                      ✗
+                    </button>
+                  </td>
+                </tr>
+              ))}
+            </tbody>
+          </table>
+        </div>
+        <button
+          type="button"
+          className="btn btn-primary bg-black"
+          onClick={() => addPerson()}
+        >
+          {t('apd.keyPersonnel.addButtonText')}
+        </button>
+      </Collapsible>
+    );
+  }
+}
+
+ApdKeyPersonnel.propTypes = {
+  keyPersonnel: PropTypes.array.isRequired,
+  years: PropTypes.array.isRequired,
+  addPerson: PropTypes.func.isRequired,
+  removePerson: PropTypes.func.isRequired,
+  updateApd: PropTypes.func.isRequired
+};
+
+const mapStateToProps = ({ apd: { data } }) => {
```

Comment:
Is it weird to wish that this was at the top of the file?  It's one of the first things I look at because it tells me what props the component uses and how it gets them.  I mean, nothing to change here, just struck me that it feels weird to scroll to the bottom of the file first.  😛